src/util/triplea/MapXMLCreator/MapXMLHelper.java

```diff
@@ -0,0 +1,1008 @@
+		try {
+			final String fileName = new FileSave("Where to Save the Game XML ?", s_xmlStrings.get("info_@name") + ".xml", MapXMLCreator.s_mapFolderLocation).getPathString();
+			if (fileName == null) {
+				return;
+			}
+
+			// write the content into xml file
+			TransformerFactory transformerFactory = TransformerFactory.newInstance();
+			// transformerFactory.setAttribute("indent-number", new Integer(2));
+			Transformer transformer = transformerFactory.newTransformer();
+			transformer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "2");
+			transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+			transformer.setOutputProperty(OutputKeys.DOCTYPE_SYSTEM, "game.dtd");
+			DOMSource source = new DOMSource(getXMLDocument());
+			final File newFile = new File(fileName);
+			StreamResult result = new StreamResult(newFile);
+
+			// Output to console for testing
+			// StreamResult result = new StreamResult(System.out);
+
+			transformer.transform(source, result);
+
+			System.out.println("");
+			System.out.println("Game XML written to " + newFile.getCanonicalPath());
+		} catch (final FileNotFoundException ex) {
+			ex.printStackTrace();
+		} catch (IOException e) {
+			e.printStackTrace();
+
+		} catch (final HeadlessException ex) {
+			ex.printStackTrace();
+		} catch (final Exception ex) {
+			ex.printStackTrace();
+		}
+	}
+
+	private static Document getXMLDocument() throws ParserConfigurationException {
+		DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+		DocumentBuilder db = dbf.newDocumentBuilder();
+
+		Document doc = db.newDocument();
+		Element game = doc.createElement("game");
+		doc.appendChild(game);
+
+		Element currentElem = null;
+		String prevKeyNode = null;
+		for (final Entry<String, String> entryXMLString : s_xmlStrings.entrySet()) {
+			String key = entryXMLString.getKey();
+			String[] split = key.split("_@");
+			if (!split[0].equals(prevKeyNode)) {
+				Element parentElem = game;
+				for (String newNode : split[0].split("_")) {
+					currentElem = doc.createElement(newNode);
+					parentElem.appendChild(currentElem);
+					parentElem = currentElem;
+				}
+			}
+			currentElem.setAttribute(split[1], entryXMLString.getValue());
+			prevKeyNode = split[0];
+		}
+
+		Element map = doc.createElement("map");
+		game.appendChild(map);
+
+		boolean territoryAttatchmentNeeded = false;
+
+		map.appendChild(doc.createComment(" Territory Definitions "));
+		for (final Entry<String, HashMap<DEFINITION, Boolean>> entryTerritoryDefinition : s_territoyDefintions.entrySet()) {
+			Element territory = doc.createElement("territory");
+			territory.setAttribute("name", entryTerritoryDefinition.getKey());
+			final HashMap<DEFINITION, Boolean> territoryDefinition = entryTerritoryDefinition.getValue();
+			final int territoryDefinitionSize = territoryDefinition.size();
+			final Boolean isWater = territoryDefinition.get(DEFINITION.IS_WATER);
+			if (isWater != null && isWater == Boolean.TRUE)

```

Comment:
Can simplify from:
`if (isWater != null && isWater == Boolean.TRUE)`, to:
`if (isWater != null && isWater )`